vars/cloudFoundryDeploy.groovy

```diff
@@ -285,11 +303,40 @@ def deployMta(config) {
         }
     }
 
-    def deployStatement = "cf ${deployCommand} ${config.mtaPath} ${config.mtaDeployParameters} ${config.mtaExtensionDescriptor}"
+    def deployStatement = "cf ${deployCommand} ${config.mtaPath} ${config.mtaDeployParameters} ${mtaExtensionDescriptorParam}"
     def apiStatement = "cf api ${config.cloudFoundry.apiEndpoint} ${config.apiParameters}"
 
-    echo "[${STEP_NAME}] Deploying MTA (${config.mtaPath}) with following parameters: ${config.mtaExtensionDescriptor} ${config.mtaDeployParameters}"
-    deploy(apiStatement, deployStatement, config, null)
+    echo "[${STEP_NAME}] Deploying MTA (${config.mtaPath}) with following parameters: ${mtaExtensionDescriptorParam} ${config.mtaDeployParameters}"
+    try {
+        deploy(apiStatement, deployStatement, config, null)
+    } finally {
+        if (config.mtaExtensionCredentials && config.mtaExtensionDescriptor && fileExists(config.mtaExtensionDescriptor)) {
+            sh "mv --force ${config.mtaExtensionDescriptor}.original ${config.mtaExtensionDescriptor} || echo 'The file ${config.mtaExtensionDescriptor}.original could not be renamed.'"
```

Comment:
If you just change something if there are credentials as commented below you only need to revert in case .origin exists.

Reply:
If I am not missing anything, then line 313 is doing exactly what you are requesting right?

Reply:
yes. Missed the if statement. thx